### PR TITLE
Analysis #11

### PR DESCRIPTION
**Pull request by rpitingolo**

### CODE REVIEW

**Comment by rpitingolo** [comment]
@lhendey I added the %Clean_sales macro to this program at the beginning. I also needed to add %Sales_pars because without it there are undefined macro variables in %Clean_sales. I'm not 100% sure if %Sales_pars is behaving the way we want it to though.

**Comment by lhendey** [comment]
@maggiesu0725 - i added the restriction for clean_sales=1 .  is the rest of the code ready for review?

**Comment by maggiesu0725** [comment]
@lhendey yes the code is ready for review. Only thing is I'm not sure if I need to register the meta data, the code is in there.

File: Prog/Affordability_sf_condo.sas [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+	month_int_rate_&year. = (eff_int_rate_&year./12/100);
+
+	loan_multiplier_&year. =  month_int_rate_&year. *	( ( 1 + month_int_rate_&year. )**360	) / ( ( ( 1+ month_int_rate_&year. )**360 )-1 );
+
+
+	*calculate monthly Principal and Interest for First time Homebuyer (10% down);
+	if saleyear=&year. then PI_First&year.=saleprice*.9*loan_multiplier_&year.;
+
+	%dollar_convert(PI_first&year.,PI_first&year.r,&year.,2017);
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 - take out the code for inflation adjustment - the incomes are not inflation adjusted so we shouldn't do the prices.

File: Prog/Affordability_sf_condo.sas [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+	%mend yearloop;
+	%yearloop
+
+
+	/*Here are numbers for Average Household Income at the city level. 2012-16 ACS 
+Black	NH-White	Hispanic	AIOM	 
+61923	 165970 	92543 	 	 80028		
+numhshldsb_2012_16 numhshldsw_2012_16 numhshldsh_2012_16 numhshldsaiom_2012_16*/ 
+
+	/*80%AMI and 50% AMI as the new affordability category
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 - okay to leave as is but you could have done a loop with the years here too.

File: Prog/Affordability_sf_condo.sas [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+run;
+proc freq data=create_flags; 
+tables AMI80_first_afford AMI50_first_afford AMI80_repeat_afford AMI50_repeat_afford; 
+run;
+*proc summary at city, ward levels - so you could get % of sales in Ward 7 affordable to 
+median white family vs. median black family.;
+
+/*Proc Summary: Affordability for Owners by 80AMI and 50AMI*/
+
+proc summary data=create_flags;
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725  - don't need city variable (they are all DC).

File: Prog/Affordability_sf_condo.sas [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+	run;
+
+	
+proc export data=sales_afford_SF_Condo
+	outfile="D:\DCDATA\Libraries\DMPED\Prog\sf_condo_tabs_aff.csv"
+	dbms=csv replace;
+	run;
+
+	
+	** Register metadata **;
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 you can delete this code for registering the metadata. we don't need to create any permanent data sets here.

File: Prog/Affordability_sf_condo.sas (hunk 3 above)
Comment (lhendey) [reply Prog/Affordability_sf_condo.sas]: @maggiesu0725 please add a proc sort before the proc summaries and change the class statement to a by statement. then you won't have all the extra groupings.

File: Prog/Affordability_sf_condo.sas (hunk 1 above)
Comment (lhendey) [reply Prog/Affordability_sf_condo.sas]: take out code for inflation adjustment

File: Prog/Affordability_sf_condo.sas [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+%Dc_update_meta_file(
+      ds_lib=DMPED,
+      ds_name=sales_afford_SF_Condo,
+      creator_process=Affordability_sf_condo.sas,
+      restrictions=None,
+      revisions=New file.
+      )
+
+data wardonly;
+	set DMPED.sales_afford_SF_Condo (where=(ward2012~=" ") keep=ward2012 pct:);
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 - just keep it a temporary file

File: Prog/Affordability_sf_condo.sas [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+	/*add code for repeat buyer affordability*
+	PctAffordRepeat_80AMI=80AMI_repeat_afford/total_sales*100; 
+*/
+	label PctAffordFirst_80AMI="Pct. of SF/Condo Sales 2000-17 Affordable to 80% AMI and 50% AMI"
+			;
+	run;
+
+	
+proc export data=sales_afford_SF_Condo
+	outfile="D:\DCDATA\Libraries\DMPED\Prog\sf_condo_tabs_aff.csv"
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 your outfile statement should use the default path (d when you are running off of d and L on L see example: 
outfile="**&_dcdata_default_path**\Requests\Prog\2018\washington region feature\county_sales_adjusted.csv"

File: Prog/Affordability_sf_condo.sas (hunk 5 above)
Comment (lhendey) [reply Prog/Affordability_sf_condo.sas]: @maggiesu0725 - you can transpose the city and ward together - should be easier once you replace the class statements with the by statements

File: Prog/Affordability_sf_condo.sas [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+	run;
+proc sort data=city_long;
+by _name_;
+proc sort data=ward_long;
+by _name_; 
+
+	data output_table;
+	merge city_long ward_long;
+	by _name_;
+	run;
```
Comment (lhendey) [inline Prog/Affordability_sf_condo.sas]: @maggiesu0725 add csv output please.

**Comment by maggiesu0725** [comment]
@lhendey took out inflation adjustment and city variable. Would love to learn how to put yearly 80% AMI into a loop, tried to but failed.

**Comment by lhendey** [comment]
@maggiesu0725 - i can write some if you want and send it to you.

**Comment by maggiesu0725** [comment]
that would be great! Thanks @lhendey

**Comment by lhendey** [comment]
@maggiesu0725  probably needs a little testing but something like:

 
%let areamedian=82800 85600 91500 84800 $85400 89300 90300 94500 99000 102700 103500 106100 107500 107300 107000 109200 108600 110300;
%let yearlist=2000 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017;
%macro calc_affordbyAMI;

%do i=1 %to 18;

	%let ami=%scan(&areamedian,&i.," "); 
	%let yr=%scan(&yearlist,&i.," "); 

	
	if saleyear =&yr. then do;
           if PITI_First <= (&ami.*0.8 / 12*.28) then AMI80_first_afford=1; else AMI80_first_afford=0; 
           if PITI_First <= (&ami.*0.5 / 12*.28) then AMI50_first_afford=1; else AMI50_first_afford=0;
		   if PITI_Repeat <= (&ami.*0.8 / 12*.28) then AMI80_repeat_afford=1; else AMI80_repeat_afford=0; 
           if PITI_Repeat <= (&ami.*0.5 / 12*.28) then AMI50_repeat_afford=1; else AMI50_repeat_afford=0; 
	end;

%end; 

%mend calc_affordbyAMI;

%calcaffordbyAMI;

**Comment by maggiesu0725** [comment]
@lhendey all comments have been addressed. The macro somehow doesn't work--- it's not generating the AMI affordability variables but there's no apparent errors when I run it either.

**Comment by lhendey** [comment]
@maggiesu0725 i fixed the macro and added code for new table. Can you add code to merge in the missing wards as you did in the other program using the MAR? (merge on after you create Sales_who_owns_SF_Condo_clean) 

also needs to be transposed at the end for the table1d format.

**Comment by lhendey** [comment]
@maggiesu0725 =is this ready for a final review so the table on box can be updated? Or did @rpitingolo review?

**Comment by lhendey** [comment]
@maggiesu0725 - this is ready to be merged, sync with L. and submit on L - then populated [table 1d. ](https://urbanorg.app.box.com/file/302739495819
) and close the issue. Please tag me when the table is ready.  cc @rpitingolo

**Pull request merged**